Fix #1771 : Method messagePrefixLength now considers other error messages like ReferenceError or TypeError, too

vhermannitk opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
In extractMessage() the error message cannot be extracted if the stack trace starts with something else then "Error". For example the message for "TypeError" won't be extracted because function messagePrefixLength() in extractMessage() returns 0, because there is no match for "match(/^Error/)".

## Motivation and Context
messagePrefixLength() just considers error messages which start with "Error". It should also consider error messages like "TypeError" or "ReferenceError".

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

sgravrock commented:
Thanks for the PR. The approach you're taking seems reasonable. However, there are a couple of problems that prevent this from being merged as-is. You've modified the build product rather than the source code (`src/core/StackTrace.js` in this case), and there are no tests for the functionality that you've added. Please review the [contributor's guide](https://github.com/jasmine/jasmine/blob/master/.github/CONTRIBUTING.md) and update accordingly.

vhermannitk commented:
@sgravrock I modified the source code instead of the build product and added some tests with my last commit. Please tell me if it is not sufficient. Sorry for the inconvenience !
